python_modules/dagster/dagster/_annotations.py

```diff
@@ -161,11 +161,13 @@ def not_deprecated_function():
         )
 
         if emit_runtime_warning:
+            stack_level = _get_warning_stacklevel(__obj)
+            subject = subject or _get_subject(__obj)
             warning_fn = lambda: deprecation_warning(
-                subject or _get_subject(__obj),
+                subject,
                 breaking_version=breaking_version,
                 additional_warn_text=additional_warn_text,
-                stacklevel=_get_warning_stacklevel(__obj),
+                stacklevel=stack_level,
```

Comment:
no logic change here right? Just making these locals for debugging or something?

Reply:
these two are intentional - slight tangent to thesis of PR but related in being unnecessary computation in the warning code path

Reply:
Oh I see you hoisted the compute out of the lambda. Missed that thanks.